lib/src/main/java/tech/relaycorp/relaydroid/messaging/ReceiveMessages.kt

```diff
@@ -0,0 +1,41 @@
+package tech.relaycorp.relaydroid.messaging
+
+import kotlinx.coroutines.flow.asFlow
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.map
+import tech.relaycorp.poweb.PoWebClient
+import tech.relaycorp.relaydroid.Relaynet
+import tech.relaycorp.relaydroid.Storage
+import tech.relaycorp.relaynet.bindings.pdc.PDCClient
+import tech.relaycorp.relaynet.bindings.pdc.Signer
+import tech.relaycorp.relaynet.bindings.pdc.StreamingMode
+import tech.relaycorp.relaynet.messages.Parcel
+
+internal class ReceiveMessages(
+    private val pdcClientBuilder: () -> PDCClient = { PoWebClient.initLocal(Relaynet.POWEB_PORT) }
+) {
+
+    fun receive() =
+        getNonceSigners()
+            .flatMapLatest { nonceSigners ->
+                pdcClientBuilder().use {
+                    it.collectParcels(nonceSigners, StreamingMode.CloseUponCompletion)
```

Comment:
I'm not sure we should force `StreamingMode.CloseUponCompletion`. I think we should either:

- Force `StreamingMode.KeepAlive`. And rely on the cancellation of the coroutine to end the underlying connection.
- Default to `StreamingMode.KeepAlive` but make it overridable. This would be useful if the app has to "come alive" periodically -- But honestly, I'm not sure that's relevant here because the gateway will wake the app up whenever there's a new message.

Thoughts?